app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/ReviewListViewModel.kt

```diff
@@ -36,4 +43,43 @@ open class ReviewListViewModel @Inject constructor(
             reviews.postValue(reviewRepo.getByReviewableId(id))
         }
     }
+
+    fun updateLikers(review: Review) {
+        if (!auth.isLoggedIn()) return
+
+        val uid = auth.getUserId()
+
+        if (review.likers.contains(uid)) {
+            reviewRepo.removeLiker(review.id, uid!!)
+        } else {
+            reviewRepo.addLiker(review.id, uid!!)
+        }
+    }
+
+    fun updateDislikers(review: Review) {
+        if (!auth.isLoggedIn()) return
+
+        val uid = auth.getUserId()
+        if (review.dislikers.contains(uid)) {
+            reviewRepo.removeDisliker(review.id, uid!!)
```

Comment:
Instead of !!, you could check if the uid is null before calling the repo function.

Reply:
It is safer indeed, thanks! I fixed it.